recipes/autoconf/all/conanfile.py

```diff
@@ -79,6 +79,10 @@ def _patch_sources(self):
         apply_conandata_patches(self)
         replace_in_file(self, os.path.join(self.source_folder, "Makefile.in"),
                         "M4 = /usr/bin/env m4", "#M4 = /usr/bin/env m4")
+        if self.settings.os == "Windows":
```

Comment:
Thanks for this one! Indeed without this change this doesn't produce a valid thing on Windows. I believe this was a pre-existing issue unrelated to the 2.0 migration - because this package has a single package_id for all platforms, the one generated on Linux by our CI works on Windows just fine.


However, precisely because there is there is a single package ID, we need to be careful about conditionals on specific platforms. 
* if this change causes the generated files to be different on Windows than when generated on Windows, we would need to change the package_id logic so that it isn't cleared
* if this change is only "when building on windows" but then the end result is the same as when built on other platforms, then the conditional check should check for the OS in `settings_build` (if present)
* otherwise, if this change is safe to be applied on _all_ platforms without affecting the result, but fixing windows, we could remove the conditional.

I'm not sure which case we're in :D  - this being the `autom4te.in` file, I'd say it's probably the second one: when building on Windows.

Reply:
Hi @jcar87,
I don't believe this change should affect the resulting files as it is only making sure that <EOL>s are stripped off of an intermediate result. I went with option (2) above, namely using settings_build, but option (3) would have probably been fine too as \R should work correctly on all platforms. I went with (2) to scope it at tightly as possible. Thanks